ceph.spec.in

```diff
@@ -1432,6 +1432,15 @@ fi
 %endif
 %{_bindir}/ceph-post-file
 %{_tmpfilesdir}/ceph-common.conf
+%if 0%{with python2}
```

Comment:
i don't think we should piggy-back python-ceph-common in ceph-common. it should be an individual python package.

Reply:
[Been there already](https://github.com/ceph/ceph/compare/8614831551594b6684760597ff14141767e5c4ba..6fe5b13eda2f14e40d751cccfd05f29ea5698fc0). Right now this is very similar to how ceph-volume is packaged in the osd package.

I had the problem that packaging the sources for python 2 and python 3 did not work as expected.

Reply:
> i don't think we should piggy-back python-ceph-common in ceph-common. it should be an individual python package.

I think we should exactly do that. We effectively need all of this code in almost every ceph deployment (not exactly right now, but in the very near future). I'd say this qualifies this to be part of the ceph-common package.

I do see the desire to logically separate domains into packages, but in the end this would just be pulled in *every* installation. My recommendation is to keep this in ceph-common.

Reply:
i agree on all these facts. but i don't think that "it's already that way so we should continue doing so". neither do i agree that "the net result is foo, so we should take the most straightforward route to foo".

let me lay out my reasons for your consideration:

- python-ceph-common is a pure python package per-se. in the long run, this package will be packaged by most mainstream distros. by packaging it separately as a arch-independent package would save the spaces and bandwidth to storing the package and transmitting it. as downstream will package Ceph for multiple architectures.
- we might need to run related script against python packages. for instance, to byte compile them and analyze their runtime dependencies. in the case of debian derivatives, they are dh_python2 and dh_python3. and these scripts spit warning messages when it's looking at binaries.
- python-ceph-common will have multiple dependencies. it would more easier for maintainers to read and analyze its dependency if it's a separated package, if they want to do this manually.

Reply:
Do you have a few minutes tomorrow to debug some strange packaging issues related to my previous attempt when creating a separate package?